pkg/purl/purl.go

```diff
@@ -120,12 +144,29 @@ func (p *PackageURL) LangType() ftypes.LangType {
 		return ftypes.Pub
 	case packageurl.TypeBitnami:
 		return ftypes.Bitnami
+	case TypeK8s:
+		if p.Namespace == "" {
+			return ftypes.K8sComponent
+		}
+		// Cloud k8s distributions, such as EKS, are not supported yet.
+		return TypeUnknown
```

Comment:
can you please explain this part ?  don't we want to add support for cloud providers in code and just add advisory data later (so we do not need to change code) ?

Reply:
I was unsure how we wanted to match cloud distributions for vulnerabilities. I've added the current idea. Please let me know if you have any feedback.
https://github.com/aquasecurity/trivy/pull/5268/commits/24f2a8b7b5c53f763d385a93050f4fcdb7759b2f